src/js/pouchdb-node.js

```diff
@@ -0,0 +1,145 @@
+/*
+
+    A node-specific extension of gpii.pouch that adds additional abilities, such as loading data from package-specific
+    paths.
+
+ */
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var fs   = require("fs");
+var os   = require("os");
+var path = require("path");
+require("../../");
+
+fluid.registerNamespace("gpii.pouch.node");
+
+/**
+ *
+ * Create our working directory if it doesn't already exist.
+ *
+ * @param that - The component itself.
+ *
+ */
+gpii.pouch.node.initDir = function (that) {
+    var fullPath = path.resolve(that.options.baseDir, that.options.dbOptions.name);
+
+    // fluid.log("pouch component '" , that.id, "' saving data to '", fullPath, "'...");
+
+    if (fs.existsSync(that.options.baseDir)) {
+        if (fs.existsSync(fullPath)) {
+            that.hasExistingDataDir = true;
+        }
+        else {
+            that.hasExistingDataDir = false;
+            fs.mkdirSync(fullPath);
+        }
+    }
+    else {
+        that.hasExistingDataDir = false;
+        fs.mkdirSync(that.options.baseDir);
+    }
+};
+
+gpii.pouch.node.makeSafePrefix = function (toResolve) {
+    return fluid.module.resolvePath(toResolve) + path.sep;
+};
+
+/**
+ *
+ * Load data from one or more package-relative paths..
+ *
+ * @param that - The component itself.
+ * @param dbPaths - A string or array representing data to be loaded.
+ * @returns {Promise} - A promise which will be resolved when all data has been loaded.
+ *
+ */
+gpii.pouch.node.loadDataFromPath = function (that, dbPaths) {
+    var promises = [];
+
+    // This instance may not actually have any data, but it should still incidate that it has finished (not) loading data.
+    if (dbPaths !== undefined) {
+        fluid.each(fluid.makeArray(dbPaths), function (singleDbPath) {
+            var data = fluid.require(fluid.module.resolvePath(singleDbPath));
```

Comment:
That crept in when I removed the previous path resolution function.  I have removed the unneeded call to `fluid.module.resolvePath`.